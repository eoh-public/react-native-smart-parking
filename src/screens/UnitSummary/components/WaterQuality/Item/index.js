import React, { memo } from 'react';
import { View, StyleSheet } from 'react-native';
import { TouchableOpacity } from 'react-native-gesture-handler';
import { IconOutline } from '@ant-design/icons-react-native';
import { useNavigation } from '@react-navigation/native';

import Text from '../../../../../commons/Text';
import { Colors, Constants } from '../../../../../configs';
import Route from '../../../../../utils/Route';

const width_item = (Constants.width - 48) / 2;

const Item = memo((props) => {
  const { svgMain, title, des, value, color, waterType } = props;

  const navigation = useNavigation();
  const goToDetail = () => {
    navigation.navigate(Route.WaterQualityGuide, { waterType });
  };

  return (
    <View style={styles.container}>
      <View size={14} color={Colors.Gray8} style={styles.textTitleX}>
        <Text>{title}</Text>
        {waterType && (
          <TouchableOpacity onPress={goToDetail}>
            <IconOutline
              style={styles.row}
              name="info-circle"
              size={14}
              color={Colors.Black}
            />
          </TouchableOpacity>
        )}
      </View>
      <Text size={24} color={color || Colors.Gray9} style={styles.textValue}>
        {value}
      </Text>
      <Text size={12} color={Colors.Gray8}>
        {des}
      </Text>
      <View style={styles.boxSvg}>
        <Text>{svgMain}</Text>
      </View>
    </View>
  );
});
export default Item;

const styles = StyleSheet.create({
  container: {
    padding: 16,
    shadowColor: Colors.Shadow,
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.1,
    elevation: 5,
    backgroundColor: Colors.White,
    width: width_item,
    marginBottom: 16,
    borderRadius: 5,
  },
  textTitleX: {
    width: '100%',
    marginBottom: 16,
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
  textTitle: {
    marginBottom: 16,
  },
  textValue: {
    marginBottom: 8,
  },
  boxSvg: {
    position: 'absolute',
    right: 8,
    top: 8,
  },
  row: {
    flexDirection: 'row',
  },
});
