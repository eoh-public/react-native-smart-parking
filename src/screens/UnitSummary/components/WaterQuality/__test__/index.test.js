import React from 'react';
import { TouchableOpacity } from 'react-native-gesture-handler';
import { act, create } from 'react-test-renderer';
import Routes from '../../../../../utils/Route';
import WaterQuality from '../index';
import { Section } from '../../../../../commons';

const mockedNavigate = jest.fn();

jest.mock('@react-navigation/native', () => {
  return {
    ...jest.requireActual('@react-navigation/native'),
    useNavigation: () => ({
      navigate: mockedNavigate,
    }),
  };
});

describe('Test WaterQualityGuide', () => {
  let data;

  beforeEach(() => {
    Date.now = jest.fn(() => new Date('2021-01-24T12:00:00.000Z'));
    data = {
      summaryDetail: {
        ph_id: 1,
        tur_id: 1,
        clo_id: 1,
        tur_value: 1,
        tur_color: '',
        tur_status: '',
      },
    };
  });
  let tree;

  test('render WaterQualityGuide', async () => {
    act(() => {
      tree = create(<WaterQuality {...data} />);
    });
    expect(tree.toJSON()).toMatchSnapshot();
    const instance = tree.root;
    const sections = instance.findAllByType(Section);
    expect(sections).toHaveLength(2);
  });

  test('render WaterQuality without showBoxHistory', async () => {
    data.summaryDetail.ph_id = 0;
    data.summaryDetail.tur_id = 0;
    data.summaryDetail.clo_id = 0;
    act(() => {
      tree = create(<WaterQuality {...data} />);
    });
    expect(tree.toJSON()).toMatchSnapshot();
    const instance = tree.root;
    const sections = instance.findAllByType(Section);
    expect(sections).toHaveLength(1);
  });

  test('onClickItem waterType', async () => {
    act(() => {
      tree = create(<WaterQuality {...data} />);
    });
    const instance = tree.root;
    const buttons = instance.findAllByType(TouchableOpacity);
    expect(buttons).toHaveLength(3);

    act(() => {
      buttons[0].props.onPress();
    });
    expect(mockedNavigate).toHaveBeenCalledTimes(1);
    expect(mockedNavigate).toHaveBeenCalledWith(Routes.WaterQualityGuide, {
      waterType: 'turbidity',
    });

    act(() => {
      buttons[1].props.onPress();
    });
    expect(mockedNavigate).toHaveBeenCalledTimes(2);
    expect(mockedNavigate).toHaveBeenCalledWith(Routes.WaterQualityGuide, {
      waterType: 'ph',
    });

    act(() => {
      buttons[2].props.onPress();
    });
    expect(mockedNavigate).toHaveBeenCalledTimes(3);
    expect(mockedNavigate).toHaveBeenCalledWith(Routes.WaterQualityGuide, {
      waterType: 'clo',
    });
  });
});
