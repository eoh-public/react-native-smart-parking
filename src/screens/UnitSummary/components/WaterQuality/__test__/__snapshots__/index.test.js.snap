// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test WaterQualityGuide render WaterQuality without showBoxHistory 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#FFFFFF",
        "borderColor": "#E8E8E8",
        "borderRadius": 20,
        "borderWidth": 1,
        "marginBottom": 16,
        "paddingBottom": 24,
        "paddingHorizontal": 16,
        "paddingTop": 8,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "marginTop": 16,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000",
            "fontFamily": "SFProDisplay-Semibold",
            "fontSize": undefined,
            "lineHeight": undefined,
            "textAlign": "left",
            "textDecorationLine": "none",
          },
          Object {
            "color": "#262626",
            "fontSize": 20,
          },
        ]
      }
    >
      Hôm nay, 24th January
    </Text>
  </View>
  <Text
    style={
      Array [
        Object {
          "color": "#000",
          "fontFamily": "SFProDisplay-Regular",
          "fontSize": undefined,
          "lineHeight": undefined,
          "textAlign": "left",
          "textDecorationLine": "none",
        },
        Object {
          "fontSize": 12,
          "lineHeight": 20,
          "marginBottom": 20,
        },
      ]
    }
  >
    Chất lượng nước tổng thể được dựa trên những thông số dưới đây
  </Text>
  <View
    style={
      Object {
        "flexDirection": "row",
        "flexWrap": "wrap",
        "justifyContent": "space-between",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#FFFFFF",
          "borderRadius": 5,
          "elevation": 5,
          "marginBottom": 16,
          "padding": 16,
          "shadowColor": "#303133",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.1,
          "width": 351,
        }
      }
    >
      <View
        color="#595959"
        size={14}
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginBottom": 16,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#000",
                "fontFamily": "SFProDisplay-Regular",
                "fontSize": undefined,
                "lineHeight": undefined,
                "textAlign": "left",
                "textDecorationLine": "none",
              },
              Object {},
            ]
          }
        >
          Độ đục
        </Text>
        <RNGestureHandlerButton
          collapsable={false}
          onGestureEvent={[Function]}
          onGestureHandlerEvent={[Function]}
          onGestureHandlerStateChange={[Function]}
          onHandlerStateChange={[Function]}
          rippleColor={0}
        >
          <View
            accessible={true}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontFamily": "antoutline",
                    "fontSize": 14,
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {
                    "flexDirection": "row",
                  },
                ]
              }
            >
              
            </Text>
          </View>
        </RNGestureHandlerButton>
      </View>
      <Text
        style={
          Array [
            Object {
              "color": "#262626",
              "fontFamily": "SFProDisplay-Regular",
              "fontSize": 24,
              "lineHeight": 24,
              "textAlign": "left",
              "textDecorationLine": "none",
            },
            Object {
              "marginBottom": 8,
            },
          ]
        }
      >
        1
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#595959",
              "fontFamily": "SFProDisplay-Regular",
              "fontSize": 12,
              "lineHeight": 12,
              "textAlign": "left",
              "textDecorationLine": "none",
            },
            Object {},
          ]
        }
      >
        
      </Text>
      <View
        style={
          Object {
            "position": "absolute",
            "right": 8,
            "top": 8,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#000",
                "fontFamily": "SFProDisplay-Regular",
                "fontSize": undefined,
                "lineHeight": undefined,
                "textAlign": "left",
                "textDecorationLine": "none",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#FFFFFF",
          "borderRadius": 5,
          "elevation": 5,
          "marginBottom": 16,
          "padding": 16,
          "shadowColor": "#303133",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.1,
          "width": 351,
        }
      }
    >
      <View
        color="#595959"
        size={14}
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginBottom": 16,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#000",
                "fontFamily": "SFProDisplay-Regular",
                "fontSize": undefined,
                "lineHeight": undefined,
                "textAlign": "left",
                "textDecorationLine": "none",
              },
              Object {},
            ]
          }
        >
          pH
        </Text>
        <RNGestureHandlerButton
          collapsable={false}
          onGestureEvent={[Function]}
          onGestureHandlerEvent={[Function]}
          onGestureHandlerStateChange={[Function]}
          onHandlerStateChange={[Function]}
          rippleColor={0}
        >
          <View
            accessible={true}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontFamily": "antoutline",
                    "fontSize": 14,
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {
                    "flexDirection": "row",
                  },
                ]
              }
            >
              
            </Text>
          </View>
        </RNGestureHandlerButton>
      </View>
      <Text
        style={
          Array [
            Object {
              "color": "#262626",
              "fontFamily": "SFProDisplay-Regular",
              "fontSize": 24,
              "lineHeight": 24,
              "textAlign": "left",
              "textDecorationLine": "none",
            },
            Object {
              "marginBottom": 8,
            },
          ]
        }
      />
      <Text
        style={
          Array [
            Object {
              "color": "#595959",
              "fontFamily": "SFProDisplay-Regular",
              "fontSize": 12,
              "lineHeight": 12,
              "textAlign": "left",
              "textDecorationLine": "none",
            },
            Object {},
          ]
        }
      />
      <View
        style={
          Object {
            "position": "absolute",
            "right": 8,
            "top": 8,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#000",
                "fontFamily": "SFProDisplay-Regular",
                "fontSize": undefined,
                "lineHeight": undefined,
                "textAlign": "left",
                "textDecorationLine": "none",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#FFFFFF",
          "borderRadius": 5,
          "elevation": 5,
          "marginBottom": 16,
          "padding": 16,
          "shadowColor": "#303133",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.1,
          "width": 351,
        }
      }
    >
      <View
        color="#595959"
        size={14}
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginBottom": 16,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#000",
                "fontFamily": "SFProDisplay-Regular",
                "fontSize": undefined,
                "lineHeight": undefined,
                "textAlign": "left",
                "textDecorationLine": "none",
              },
              Object {},
            ]
          }
        >
          Clo dư
        </Text>
        <RNGestureHandlerButton
          collapsable={false}
          onGestureEvent={[Function]}
          onGestureHandlerEvent={[Function]}
          onGestureHandlerStateChange={[Function]}
          onHandlerStateChange={[Function]}
          rippleColor={0}
        >
          <View
            accessible={true}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontFamily": "antoutline",
                    "fontSize": 14,
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {
                    "flexDirection": "row",
                  },
                ]
              }
            >
              
            </Text>
          </View>
        </RNGestureHandlerButton>
      </View>
      <Text
        style={
          Array [
            Object {
              "color": "#262626",
              "fontFamily": "SFProDisplay-Regular",
              "fontSize": 24,
              "lineHeight": 24,
              "textAlign": "left",
              "textDecorationLine": "none",
            },
            Object {
              "marginBottom": 8,
            },
          ]
        }
      />
      <Text
        style={
          Array [
            Object {
              "color": "#595959",
              "fontFamily": "SFProDisplay-Regular",
              "fontSize": 12,
              "lineHeight": 12,
              "textAlign": "left",
              "textDecorationLine": "none",
            },
            Object {},
          ]
        }
      />
      <View
        style={
          Object {
            "position": "absolute",
            "right": 8,
            "top": 8,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#000",
                "fontFamily": "SFProDisplay-Regular",
                "fontSize": undefined,
                "lineHeight": undefined,
                "textAlign": "left",
                "textDecorationLine": "none",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#FFFFFF",
          "borderRadius": 5,
          "elevation": 5,
          "marginBottom": 16,
          "padding": 16,
          "shadowColor": "#303133",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.1,
          "width": 351,
        }
      }
    >
      <View
        color="#595959"
        size={14}
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginBottom": 16,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#000",
                "fontFamily": "SFProDisplay-Regular",
                "fontSize": undefined,
                "lineHeight": undefined,
                "textAlign": "left",
                "textDecorationLine": "none",
              },
              Object {},
            ]
          }
        >
          Nhiệt độ
        </Text>
      </View>
      <Text
        style={
          Array [
            Object {
              "color": "#262626",
              "fontFamily": "SFProDisplay-Regular",
              "fontSize": 24,
              "lineHeight": 24,
              "textAlign": "left",
              "textDecorationLine": "none",
            },
            Object {
              "marginBottom": 8,
            },
          ]
        }
      />
      <Text
        style={
          Array [
            Object {
              "color": "#595959",
              "fontFamily": "SFProDisplay-Regular",
              "fontSize": 12,
              "lineHeight": 12,
              "textAlign": "left",
              "textDecorationLine": "none",
            },
            Object {},
          ]
        }
      >
        
      </Text>
      <View
        style={
          Object {
            "position": "absolute",
            "right": 8,
            "top": 8,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#000",
                "fontFamily": "SFProDisplay-Regular",
                "fontSize": undefined,
                "lineHeight": undefined,
                "textAlign": "left",
                "textDecorationLine": "none",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`Test WaterQualityGuide render WaterQualityGuide 1`] = `
Array [
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#FFFFFF",
          "borderColor": "#E8E8E8",
          "borderRadius": 20,
          "borderWidth": 1,
          "marginBottom": 16,
          "paddingBottom": 24,
          "paddingHorizontal": 16,
          "paddingTop": 8,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "marginTop": 16,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#000",
              "fontFamily": "SFProDisplay-Semibold",
              "fontSize": undefined,
              "lineHeight": undefined,
              "textAlign": "left",
              "textDecorationLine": "none",
            },
            Object {
              "color": "#262626",
              "fontSize": 20,
            },
          ]
        }
      >
        Hôm nay, 24th January
      </Text>
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "#000",
            "fontFamily": "SFProDisplay-Regular",
            "fontSize": undefined,
            "lineHeight": undefined,
            "textAlign": "left",
            "textDecorationLine": "none",
          },
          Object {
            "fontSize": 12,
            "lineHeight": 20,
            "marginBottom": 20,
          },
        ]
      }
    >
      Chất lượng nước tổng thể được dựa trên những thông số dưới đây
    </Text>
    <View
      style={
        Object {
          "flexDirection": "row",
          "flexWrap": "wrap",
          "justifyContent": "space-between",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#FFFFFF",
            "borderRadius": 5,
            "elevation": 5,
            "marginBottom": 16,
            "padding": 16,
            "shadowColor": "#303133",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.1,
            "width": 351,
          }
        }
      >
        <View
          color="#595959"
          size={14}
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginBottom": 16,
              "width": "100%",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#000",
                  "fontFamily": "SFProDisplay-Regular",
                  "fontSize": undefined,
                  "lineHeight": undefined,
                  "textAlign": "left",
                  "textDecorationLine": "none",
                },
                Object {},
              ]
            }
          >
            Độ đục
          </Text>
          <RNGestureHandlerButton
            collapsable={false}
            onGestureEvent={[Function]}
            onGestureHandlerEvent={[Function]}
            onGestureHandlerStateChange={[Function]}
            onHandlerStateChange={[Function]}
            rippleColor={0}
          >
            <View
              accessible={true}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#000000",
                      "fontFamily": "antoutline",
                      "fontSize": 14,
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {
                      "flexDirection": "row",
                    },
                  ]
                }
              >
                
              </Text>
            </View>
          </RNGestureHandlerButton>
        </View>
        <Text
          style={
            Array [
              Object {
                "color": "#262626",
                "fontFamily": "SFProDisplay-Regular",
                "fontSize": 24,
                "lineHeight": 24,
                "textAlign": "left",
                "textDecorationLine": "none",
              },
              Object {
                "marginBottom": 8,
              },
            ]
          }
        >
          1
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#595959",
                "fontFamily": "SFProDisplay-Regular",
                "fontSize": 12,
                "lineHeight": 12,
                "textAlign": "left",
                "textDecorationLine": "none",
              },
              Object {},
            ]
          }
        >
          
        </Text>
        <View
          style={
            Object {
              "position": "absolute",
              "right": 8,
              "top": 8,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#000",
                  "fontFamily": "SFProDisplay-Regular",
                  "fontSize": undefined,
                  "lineHeight": undefined,
                  "textAlign": "left",
                  "textDecorationLine": "none",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#FFFFFF",
            "borderRadius": 5,
            "elevation": 5,
            "marginBottom": 16,
            "padding": 16,
            "shadowColor": "#303133",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.1,
            "width": 351,
          }
        }
      >
        <View
          color="#595959"
          size={14}
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginBottom": 16,
              "width": "100%",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#000",
                  "fontFamily": "SFProDisplay-Regular",
                  "fontSize": undefined,
                  "lineHeight": undefined,
                  "textAlign": "left",
                  "textDecorationLine": "none",
                },
                Object {},
              ]
            }
          >
            pH
          </Text>
          <RNGestureHandlerButton
            collapsable={false}
            onGestureEvent={[Function]}
            onGestureHandlerEvent={[Function]}
            onGestureHandlerStateChange={[Function]}
            onHandlerStateChange={[Function]}
            rippleColor={0}
          >
            <View
              accessible={true}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#000000",
                      "fontFamily": "antoutline",
                      "fontSize": 14,
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {
                      "flexDirection": "row",
                    },
                  ]
                }
              >
                
              </Text>
            </View>
          </RNGestureHandlerButton>
        </View>
        <Text
          style={
            Array [
              Object {
                "color": "#262626",
                "fontFamily": "SFProDisplay-Regular",
                "fontSize": 24,
                "lineHeight": 24,
                "textAlign": "left",
                "textDecorationLine": "none",
              },
              Object {
                "marginBottom": 8,
              },
            ]
          }
        />
        <Text
          style={
            Array [
              Object {
                "color": "#595959",
                "fontFamily": "SFProDisplay-Regular",
                "fontSize": 12,
                "lineHeight": 12,
                "textAlign": "left",
                "textDecorationLine": "none",
              },
              Object {},
            ]
          }
        />
        <View
          style={
            Object {
              "position": "absolute",
              "right": 8,
              "top": 8,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#000",
                  "fontFamily": "SFProDisplay-Regular",
                  "fontSize": undefined,
                  "lineHeight": undefined,
                  "textAlign": "left",
                  "textDecorationLine": "none",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#FFFFFF",
            "borderRadius": 5,
            "elevation": 5,
            "marginBottom": 16,
            "padding": 16,
            "shadowColor": "#303133",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.1,
            "width": 351,
          }
        }
      >
        <View
          color="#595959"
          size={14}
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginBottom": 16,
              "width": "100%",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#000",
                  "fontFamily": "SFProDisplay-Regular",
                  "fontSize": undefined,
                  "lineHeight": undefined,
                  "textAlign": "left",
                  "textDecorationLine": "none",
                },
                Object {},
              ]
            }
          >
            Clo dư
          </Text>
          <RNGestureHandlerButton
            collapsable={false}
            onGestureEvent={[Function]}
            onGestureHandlerEvent={[Function]}
            onGestureHandlerStateChange={[Function]}
            onHandlerStateChange={[Function]}
            rippleColor={0}
          >
            <View
              accessible={true}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#000000",
                      "fontFamily": "antoutline",
                      "fontSize": 14,
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {
                      "flexDirection": "row",
                    },
                  ]
                }
              >
                
              </Text>
            </View>
          </RNGestureHandlerButton>
        </View>
        <Text
          style={
            Array [
              Object {
                "color": "#262626",
                "fontFamily": "SFProDisplay-Regular",
                "fontSize": 24,
                "lineHeight": 24,
                "textAlign": "left",
                "textDecorationLine": "none",
              },
              Object {
                "marginBottom": 8,
              },
            ]
          }
        />
        <Text
          style={
            Array [
              Object {
                "color": "#595959",
                "fontFamily": "SFProDisplay-Regular",
                "fontSize": 12,
                "lineHeight": 12,
                "textAlign": "left",
                "textDecorationLine": "none",
              },
              Object {},
            ]
          }
        />
        <View
          style={
            Object {
              "position": "absolute",
              "right": 8,
              "top": 8,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#000",
                  "fontFamily": "SFProDisplay-Regular",
                  "fontSize": undefined,
                  "lineHeight": undefined,
                  "textAlign": "left",
                  "textDecorationLine": "none",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#FFFFFF",
            "borderRadius": 5,
            "elevation": 5,
            "marginBottom": 16,
            "padding": 16,
            "shadowColor": "#303133",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.1,
            "width": 351,
          }
        }
      >
        <View
          color="#595959"
          size={14}
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginBottom": 16,
              "width": "100%",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#000",
                  "fontFamily": "SFProDisplay-Regular",
                  "fontSize": undefined,
                  "lineHeight": undefined,
                  "textAlign": "left",
                  "textDecorationLine": "none",
                },
                Object {},
              ]
            }
          >
            Nhiệt độ
          </Text>
        </View>
        <Text
          style={
            Array [
              Object {
                "color": "#262626",
                "fontFamily": "SFProDisplay-Regular",
                "fontSize": 24,
                "lineHeight": 24,
                "textAlign": "left",
                "textDecorationLine": "none",
              },
              Object {
                "marginBottom": 8,
              },
            ]
          }
        />
        <Text
          style={
            Array [
              Object {
                "color": "#595959",
                "fontFamily": "SFProDisplay-Regular",
                "fontSize": 12,
                "lineHeight": 12,
                "textAlign": "left",
                "textDecorationLine": "none",
              },
              Object {},
            ]
          }
        >
          
        </Text>
        <View
          style={
            Object {
              "position": "absolute",
              "right": 8,
              "top": 8,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#000",
                  "fontFamily": "SFProDisplay-Regular",
                  "fontSize": undefined,
                  "lineHeight": undefined,
                  "textAlign": "left",
                  "textDecorationLine": "none",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
      </View>
    </View>
  </View>,
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#FFFFFF",
          "borderColor": "#E8E8E8",
          "borderRadius": 20,
          "borderWidth": 1,
          "marginBottom": 16,
          "paddingBottom": 24,
          "paddingHorizontal": 16,
          "paddingTop": 8,
        },
        undefined,
      ]
    }
  >
    <View>
      <View
        style={
          Object {
            "paddingTop": 16,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#262626",
                "fontFamily": "SFProDisplay-Semibold",
                "fontSize": 20,
                "lineHeight": 20,
                "textAlign": "left",
                "textDecorationLine": "none",
              },
              Object {},
            ]
          }
        >
          Lịch sử
        </Text>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "marginTop": 16,
                "width": "100%",
              },
              undefined,
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#8C8C8C",
                  "fontFamily": "SFProDisplay-Regular",
                  "fontSize": 12,
                  "lineHeight": 12,
                  "textAlign": "left",
                  "textDecorationLine": "none",
                },
                Object {},
              ]
            }
          >
            từ
          </Text>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "borderColor": "#D9D9D9",
                "borderRadius": 2,
                "borderWidth": 1,
                "flexDirection": "row",
                "justifyContent": "space-between",
                "marginHorizontal": 8,
                "opacity": 1,
                "paddingLeft": 12,
                "paddingRight": 8,
                "paddingVertical": 8,
                "width": 124,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#595959",
                    "fontFamily": "SFProDisplay-Regular",
                    "fontSize": 12,
                    "lineHeight": 20,
                    "textAlign": "left",
                    "textDecorationLine": "none",
                  },
                  Object {
                    "marginRight": 4,
                  },
                ]
              }
            >
              23.01.21 12:00
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "#595959",
                    "fontFamily": "antfill",
                    "fontSize": 16,
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  undefined,
                ]
              }
            >
              
            </Text>
          </View>
          <Text
            style={
              Array [
                Object {
                  "color": "#8C8C8C",
                  "fontFamily": "SFProDisplay-Regular",
                  "fontSize": 12,
                  "lineHeight": 12,
                  "textAlign": "left",
                  "textDecorationLine": "none",
                },
                Object {},
              ]
            }
          >
            đến
          </Text>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "borderColor": "#D9D9D9",
                "borderRadius": 2,
                "borderWidth": 1,
                "flexDirection": "row",
                "justifyContent": "space-between",
                "marginHorizontal": 8,
                "opacity": 1,
                "paddingLeft": 12,
                "paddingRight": 8,
                "paddingVertical": 8,
                "width": 124,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#595959",
                    "fontFamily": "SFProDisplay-Regular",
                    "fontSize": 12,
                    "lineHeight": 20,
                    "textAlign": "left",
                    "textDecorationLine": "none",
                  },
                  Object {
                    "marginRight": 4,
                  },
                ]
              }
            >
              24.01.21 12:00
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "#595959",
                    "fontFamily": "antfill",
                    "fontSize": 16,
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  undefined,
                ]
              }
            >
              
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "marginLeft": -8,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "height": 300,
              },
              Object {
                "width": 750,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#FFFFFF",
                "flex": 1,
                "justifyContent": "center",
                "marginRight": 24,
                "marginTop": 24,
              }
            }
          >
            <View />
          </View>
        </View>
      </View>
      <Modal
        animationType="none"
        hardwareAccelerated={false}
        isVisible={false}
        onHide={[Function]}
        transparent={true}
        visible={false}
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "black",
              "bottom": 0,
              "height": 1334,
              "left": 0,
              "opacity": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": 750,
            }
          }
        />
      </Modal>
      <Modal
        animationType="none"
        hardwareAccelerated={false}
        isVisible={false}
        onHide={[Function]}
        transparent={true}
        visible={false}
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "black",
              "bottom": 0,
              "height": 1334,
              "left": 0,
              "opacity": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": 750,
            }
          }
        />
      </Modal>
    </View>
  </View>,
]
`;
